src/ios/CDVReachabilityManager.m

```diff
@@ -0,0 +1,210 @@
+
+@implementation ProxyInfo
+- (BOOL)isEqual:(id)object {
+    ProxyInfo *toCompare = (ProxyInfo *)object;
+    BOOL isEqual = NO;
+    if (toCompare) {
+        isEqual = [toCompare.type isEqualToString:self.type] && (toCompare.host == self.host || [toCompare.host isEqualToString:self.host]) && (toCompare.port == self.port || [toCompare.port isEqualToString:self.port]) && (toCompare.urlKey == self.urlKey || [toCompare.urlKey isEqualToString:self.urlKey]);
+    }
+    return isEqual;
+}
+@end
+
+@implementation NetworkInfo
+@end
+
+@implementation ReachabilityManager
+
+
+- (void)start:(NSURL*)url :(SPNetworkInfoBlock)completionHandler {
+    self.proxyUrl = url;
+    self.networkInfoBlock = completionHandler;
+    [self startReachabilityListener];
+}
+
+- (void)stop {
+    [self stopReachabilityListener];
+}
+
+- (void)startReachabilityListener {
+    if (self.reachabilityListenerStarted == YES) {
+        return;
+    }
+    self.reachabilityListenerStarted = YES;
+
+    self.reach = [CPReachability reachabilityForInternetConnection];
+
+    __weak ReachabilityManager* weakSelf = self;
+    self.reach.reachableBlock = ^(CPReachability*reach)
+    {
+        NSLog(@"===qqq REACHABLE!");
+        [weakSelf checkNetworkData];
+    };
+
+    self.reach.unreachableBlock = ^(CPReachability*reach)
+    {
+        NSLog(@"===qqq UNREACHABLE!");
+    };
+    [self.reach startNotifier];
+    [self checkNetworkData];
+
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(onPause) name:UIApplicationDidEnterBackgroundNotification object:nil];
+    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(onResume) name:UIApplicationWillEnterForegroundNotification object:nil];
+}
+
+- (void)stopReachabilityListener {
+    if (self.reachabilityListenerStarted == NO) {
+        return;
+    }
+    self.reachabilityListenerStarted = NO;
+    [[NSNotificationCenter defaultCenter] removeObserver:self];
+    [self.reach stopNotifier];
+    self.reach = NULL;
+    self.networkInfoBlock = NULL;
+    self.proxyUrl = NULL;
+}
+
+- (void)onPause
+{
+    if (self.reachabilityListenerStarted == NO) {
+        return;
+    }
+    [self.reach stopNotifier];
+}
+
+- (void)onResume
+{
+    if (self.reachabilityListenerStarted == NO) {
+        return;
+    }
+    [self.reach startNotifier];
+    [self checkNetworkData];
+}
+
+- (void) checkNetworkData {
+
+    if (!self.reachabilityListenerStarted) {
+        return;
+    }
+
+    if (!self.reach || !self.reach.isReachable) {
+        return;
+    }
+
+    NSArray<NetworkAddress *> * currentNetworkAddresses = [ReachabilityIPHelper getNetworkAddress];
+    NSArray<ProxyInfo *> * currentProxiesInfo = [self getProxySettings];
+    NSString* currentTransport = [self getCurrentNetworkStatus];
+
+    BOOL isNetworkAddressChanged = NO;
+    BOOL isProxyInfoChanged = NO;
+    BOOL isTransportChanged = NO;
+
+
+    if (![self.prevNetworkAddress isEqualToArray:currentNetworkAddresses]) {
```

Comment:
Too difficult construction. This is oneline sentence:
`isNetworkAddressChanged = ![self.prevNetworkAddress isEqualToArray:currentNetworkAddresses]`

Reply:
done